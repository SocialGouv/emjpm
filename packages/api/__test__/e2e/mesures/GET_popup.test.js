//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

test("should return mesures of user type prepose", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .get("/api/v1/mesures/popup?searchType=prepose")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();

  expect(response.status).toBe(200);
});

test("should return mesures of user type individuel", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .get("/api/v1/mesures/popup?searchType=individuel")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();

  expect(response.status).toBe(200);
});
