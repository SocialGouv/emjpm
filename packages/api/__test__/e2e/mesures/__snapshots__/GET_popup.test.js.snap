// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return mesures of user type individuel 1`] = `
Array [
  Object {
    "code_postal": "62000",
    "count": "1",
    "latitude": 50.5333,
    "longitude": 2.6333,
    "mandataire_ids": Array [
      1,
    ],
    "types": Array [
      "individuel",
    ],
  },
  Object {
    "code_postal": "93000",
    "count": "1",
    "latitude": 48.91,
    "longitude": 2.41998,
    "mandataire_ids": Array [
      1,
    ],
    "types": Array [
      "individuel",
    ],
  },
]
`;

exports[`should return mesures of user type prepose 1`] = `
Array [
  Object {
    "code_postal": "75000",
    "count": "1",
    "latitude": 48.8534,
    "longitude": 2.3488,
    "mandataire_ids": Array [
      2,
    ],
    "types": Array [
      "prepose",
    ],
  },
]
`;
